include/openthread/border_routing.h

```diff
@@ -703,6 +703,24 @@ void otBorderRoutingSetMultiAilCallback(otInstance                     *aInstanc
                                         otBorderRoutingMultiAilCallback aCallback,
                                         void                           *aContext);
 
+#if OPENTHREAD_CONFIG_NAT64_BORDER_ROUTING_ENABLE
```

Comment:
@yangsong-cnyn , we cannot and **should not** add `#if OPENTHREAD_CONFIG_*` checks to **any public or platform OT headers**. The reason is that public headers can be included from other projects (SW modules) where the OT config definitions are not available or defined.

You should instead document that this API requires this specific config. You can check other APIs to see example of how this is done:

```c++
/**
 * Iterates over the peer BRs found in the Network Data.
 *
 * Requires `OPENTHREAD_CONFIG_BORDER_ROUTING_TRACK_PEER_BR_INFO_ENABLE`.
 * ...
```